muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/ChatSyncController.java

```diff
@@ -58,279 +78,377 @@
-    }
-    if (!StringUtils.isNumeric(openfirePort)) {
-      logger.log(Level.WARNING, "Invalid openfire port, skipping room sync");
-      return;
+  private SearchProvider getSearchProvider() {
+    Iterator<SearchProvider> searchProviderIterator = searchProviders.iterator();
+    if (searchProviderIterator.hasNext()) {
+      return searchProviderIterator.next();
+    } else {
+      return null;
     }
+  }
 
-    AuthenticationToken token = new AuthenticationToken(openfireToken);
-    RestApiClient client = new RestApiClient(openfireUrl, Integer.parseInt(openfirePort, 10), token);
-    
-    
-    Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
-    
-
-    String subjectCode = courseMetaController.findSubject(workspace.getSchoolDataSource(), workspace.getSubjectIdentifier()).getCode();
-    
-    String roomName = subjectCode + workspace.getCourseNumber() + " - " + workspace.getNameExtension();
-    MUCRoomEntity chatRoomEntity = client.getChatRoom(workspace.getIdentifier());
-
-    chatRoomEntity = new MUCRoomEntity(workspace.getIdentifier(), roomName, workspace.getDescription());
-    chatRoomEntity.setPersistent(true);
-    chatRoomEntity.setLogEnabled(true);
-    client.createChatRoom(chatRoomEntity);
-    
-    workspaceChatSettingsEnabledEvent.fire(new WorkspaceChatSettingsEnabledEvent(workspace.getSchoolDataSource(), workspace.getIdentifier(), true));
+  private boolean isStudent(fi.otavanopisto.muikku.model.users.UserEntity userEntity) {
+    EnvironmentRoleEntity roleEntity = userSchoolDataIdentifierController.findUserSchoolDataIdentifierRole(userEntity);
+    return roleEntity == null || roleEntity.getArchetype() == EnvironmentRoleArchetype.STUDENT;
   }
- 
- public void syncWorkspaceUser(WorkspaceEntity workspaceEntity, SchoolDataIdentifier userIdentifier) {
-    String openfireToken = pluginSettingsController.getPluginSetting("chat", "openfireToken");
-    if (openfireToken == null) {
-      logger.log(Level.INFO, "No openfire token set, skipping room sync");
-      return;
-      }
-    String openfireUrl = pluginSettingsController.getPluginSetting("chat", "openfireUrl");
-    if (openfireUrl == null) {
-      logger.log(Level.INFO, "No openfire url set, skipping room sync");
-      return;
-    }
-    String openfirePort = pluginSettingsController.getPluginSetting("chat", "openfirePort");
-    if (openfirePort == null) {
-      logger.log(Level.INFO, "No openfire port set, skipping room sync");
+  
+  private void addAdminsAsRoomOwners(RestApiClient client, MUCRoomEntity mucRoomEntity) {
+    SearchProvider searchProvider = getSearchProvider();
+    if (searchProvider == null) {
+      logger.severe("ElasticSearch missing");
       return;
     }
-    if (!StringUtils.isNumeric(openfirePort)) {
-      logger.log(Level.WARNING, "Invalid openfire port, skipping room sync");
-      return;
+    List<OrganizationEntity> organizations = organizationEntityController.listLoggedUserOrganizations();
```

Comment:
Is using logged user here a correct call?

Reply:
We have to limit the search to return admins and study programmer leaders of only "our own" organization, so the code assumes the organization is that of the logged-in user doing stuff that causes this method to be called (basically whenever a new chat room is created).